app/notifications/notifications_ses_callback.py

```diff
@@ -44,6 +57,50 @@ def _determine_provider_response(ses_message):
     return None
 
 
+def _determine_bounce_response(ses_message):
+    if ses_message["notificationType"] != "Bounce":
+        return None
+
+    bounce_type = ses_message["bounce"]["bounceType"]
+    bounce_subtype = ses_message["bounce"]["bounceSubType"]
+
+    bounce_response = {
+        "feedback_type": "",
+        "feedback_subtype": "",
+        "ses_feedback_id": ses_message["bounce"]["feedbackId"],
+        "ses_feedback_date": ses_message["bounce"]["timestamp"],
+    }
+
+    # See https://docs.aws.amazon.com/ses/latest/dg/notification-contents.html#bounce-types for all bounce types
+    if bounce_type == "Undetermined":  # treat this as a soft bounce since we don't know what went wrong
+        bounce_response["feedback_type"] = NOTIFICATION_SOFT_BOUNCE
+        bounce_response["feedback_subtype"] = NOTIFICATION_SOFT_GENERAL
+    elif bounce_type == "Permanent":
+        bounce_response["feedback_type"] = NOTIFICATION_HARD_BOUNCE
+        if bounce_subtype == "General":
+            bounce_response["feedback_subtype"] = NOTIFICATION_HARD_GENERAL
+        if bounce_subtype == "NoEmail":
+            bounce_response["feedback_subtype"] = NOTIFICATION_HARD_NOEMAIL
+        if bounce_subtype == "Suppressed":
+            bounce_response["feedback_subtype"] = NOTIFICATION_HARD_SUPPRESSED
+        if bounce_subtype == "OnAccountSuppressionList":
+            bounce_response["feedback_subtype"] = NOTIFICATION_HARD_ONACCOUNTSUPPRESSIONLIST
+    elif bounce_type == "Transient":
+        bounce_response["feedback_type"] = NOTIFICATION_SOFT_BOUNCE
+        if bounce_subtype == "General":
+            bounce_response["feedback_subtype"] = NOTIFICATION_SOFT_GENERAL
+        if bounce_subtype == "MailboxFull":
+            bounce_response["feedback_subtype"] = NOTIFICATION_SOFT_MAILBOXFULL
+        if bounce_subtype == "MessageTooLarge":
+            bounce_response["feedback_subtype"] = NOTIFICATION_SOFT_MESSAGETOOLARGE
+        if bounce_subtype == "ContentRejected":
+            bounce_response["feedback_subtype"] = NOTIFICATION_SOFT_CONTENTREJECTED
+        if bounce_subtype == "AttachmentRejected":
+            bounce_response["feedback_subtype"] = NOTIFICATION_SOFT_ATTACHMENTREJECTED
```

Comment:
Should we cover unknown types in case the response is unexpected? or we will consider blank feedback subtype as such?

Reply:
Yes you make a good point.  It could work with the blanks now, but I think it would be best to be explicit.  I will re-work. this and add a new migration to update the `feedback_types` and `feedback_subtypes`.

Reply:
Done, have a look when you have a chance!